src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -283,16 +289,52 @@ private LSP.Diagnostic ConvertDiagnostic(Document document, SourceText text, Dia
             // would get automatically serialized.
             LSP.VSDiagnostic CreateBaseLspDiagnostic()
             {
-                return new LSP.VSDiagnostic
+                var diagnostic = new LSP.VSDiagnostic
                 {
                     Source = "Roslyn",
                     Code = diagnosticData.Id,
                     CodeDescription = ProtocolConversions.HelpLinkToCodeDescription(diagnosticData.GetValidHelpLinkUri()),
                     Message = diagnosticData.Message,
                     Severity = ConvertDiagnosticSeverity(diagnosticData.Severity),
-                    Range = ProtocolConversions.LinePositionToRange(DiagnosticData.GetLinePositionSpan(diagnosticData.DataLocation, text, useMappedSpan)),
                     Tags = ConvertTags(diagnosticData),
                 };
+                var range = GetRange(diagnosticData.DataLocation);
+                if (range != null)
+                {
+                    diagnostic.Range = range;
```

Comment:
is it bad if we get null back and we set diagnostic.Range to null?  i'm not sure how htat'c conceptually different from not setting it.